src/lib.rs

```diff
@@ -0,0 +1,230 @@
+
+                uwu_progress_bar.finish_with_message("UwU'ifying Complete!");
+                Ok(())
+            } else {
+                #[cfg(not(test))]
+                let stdout = std::io::stdout();
+                #[cfg(not(test))]
+                let mut out = BufWriter::new(stdout.lock());
+                #[cfg(test)]
+                let mut out = std::io::sink();
+                self.uwuify_sentence(self.text, &mut out)?;
+                #[cfg(not(test))]
+                out.write_all(b"\n")?;
+                Ok(())
+            }
+        } else {
+            // Handle File I/O
+            if Path::new(&self.output).exists() {
+                return Err(Error::new(
+                    std::io::ErrorKind::AlreadyExists,
+                    format!("File '{}' already exists, aborting operation", &self.output),
+                ));
+            }
+
+            let uwu_progress_bar = progress_bar!();
+            self.uwuify_sentence(
+                unsafe {
+                    std::str::from_utf8_unchecked(
+                        memmap::Mmap::map(&File::open(&self.input)?)?.as_ref(),
+                    )
+                },
+                &mut BufWriter::new(File::create(&self.output)?),
+            )?;
+            uwu_progress_bar.finish_with_message("UwU'ifying Complete!");
+            Ok(())
+        }
+    }
+
+    fn uwuify_sentence<T: Write>(&mut self, text: &str, out: &mut T) -> Result<(), std::io::Error> {
+        text.lines().try_for_each(|line| {
+            line.split_whitespace()
+                .map(|f| f.as_bytes())
+                .try_for_each(|word| {
+                    let random_value = self.floating_rng.gen_range(0.0..1.0);
+
+                    if random_value <= self.faces {
+                        out.write_all(FACES[self.int_rng.gen_range(0..FACES_SIZE)])?;
+                        out.write_all(b" ")?;
+                    } else if random_value <= self.actions {
+                        out.write_all(ACTIONS[self.int_rng.gen_range(0..ACTIONS_SIZE)])?;
+                        out.write_all(b" ")?;
+                    } else if random_value <= self.stutters {
+                        (0..self.int_rng.gen_range(1..2))
+                            .into_iter()
+                            .try_for_each(|_| {
+                                match word[0] {
+                                    b'L' | b'R' => out.write_all(b"W"),
+                                    b'l' | b'r' => out.write_all(b"w"),
+                                    b => out.write_all(&[b]),
+                                }?;
+                                out.write_all(b"-")
+                            })?;
+                    }
+
+                    if self
+                        .linkify
+                        .links(unsafe { std::str::from_utf8_unchecked(word) })
+                        .count()
+                        > 0
```

Comment:
This is fairly hard to read? Maybe split `self.linkify.links(...` into it's own variable?

Reply:
Nah, inlining the variable into the if statement like this ensures that the count produced by it is dropped as soon as the `if` statement ends

Reply:
Ah right sure, that makes